torch/csrc/distributed/c10d/intra_node_comm.cpp

```diff
@@ -151,6 +173,14 @@ bool IntraNodeComm::rendezvous() {
   gethostname(devInfo.hostname, sizeof(devInfo.hostname));
   devInfo.deviceIdx = deviceIdx_;
 
+#if defined(USE_ROCM)
+  auto ret = rsmi_init(0);
```

Comment:
What is rsmi? Is it similar to nvml?

Reply:
yes, this is how we determine gpu topology.